open-machine-learning-jupyter-book/ml-advanced/clustering/introduction-to-clustering.md

```diff
@@ -0,0 +1,323 @@
+| Mean-shift                   | many, uneven clusters, inductive                                       |
+| Spectral clustering          | few, even clusters, transductive                                       |
+| Ward hierarchical clustering | many, constrained clusters, transductive                               |
+| Agglomerative clustering     | many, constrained, non Euclidean distances, transductive               |
+| DBSCAN                       | non-flat geometry, uneven clusters, transductive                       |
+| OPTICS                       | non-flat geometry, uneven clusters with variable density, transductive |
+| Gaussian mixtures            | flat geometry, inductive                                               |
+| BIRCH                        | large dataset with outliers, inductive                                 |
+
+> 🎓 How we create clusters has a lot to do with how we gather up the data points into groups. Let's unpack some vocabulary:
+>
+> 🎓 'Transductive' vs. 'inductive'
+> 
+> Transductive inference is derived from observed training cases that map to specific test cases. Inductive inference is derived from training cases that map to general rules which are only then applied to test cases. 
+> 
+> An example: Imagine you have a dataset that is only partially  labelled. Some things are 'records', some 'cds', and some are blank. Your job is to provide labels for the blanks. If you choose an inductive approach, you'd train a model looking for 'records' and 'cds', and apply those labels to your unlabeled data. This approach will have trouble classifying things that are actually 'cassettes'. A transductive approach, on the other hand, handles this unknown data more effectively as it works to group similar items together and then applies a label to a group. In this case, clusters might reflect 'round musical things' and 'square musical things'. 
+> 
+> 🎓 'Non-flat' vs. 'flat' geometry
+> 
+> Derived from mathematical terminology, non-flat vs. flat geometry refers to the measure of distances between points by either 'flat' (Euclidean) or 'non-flat' (non-Euclidean) geometrical methods. 
+>
+>'Flat' in this context refers to Euclidean geometry (parts of which are taught as 'plane' geometry), and non-flat refers to non-Euclidean geometry. What does geometry have to do with machine learning? Well, as two fields that are rooted in mathematics, there must be a common way to measure distances between points in clusters, and that can be done in a 'flat' or 'non-flat' way, depending on the nature of the data. Euclidean distances are measured as the length of a line segment between two points. Non-Euclidean distances are measured along a curve. If your data, visualized, seems to not exist on a plane, you might need to use a specialized algorithm to handle it.
+>
+![Flat vs Nonflat Geometry Infographic](../../../images/clustering/flat-nonflat.png)
+> Infographic by Dasani Madipalli
```

Comment:
As a better way to refer the image, you can use {figure} with title and author info.

https://jupyterbook.org/en/stable/content/figures.html#figures

Reply:
Please apply this to all the new added images. Also, you can find examples by searching {figure} in the code base.